src/promptflow/promptflow/batch/_base_executor_proxy.py

```diff
@@ -134,6 +147,27 @@ async def ensure_executor_health(self):
             retry_count += 1
         raise ExecutorServiceUnhealthy(f"{EXECUTOR_UNHEALTHY_MESSAGE}. Please resubmit your flow and try again.")
 
+    async def _check_health(self):
+        try:
+            health_url = self.api_endpoint + "/health"
+            async with httpx.AsyncClient() as client:
+                response = await client.get(health_url)
+            if response.status_code != 200:
+                bulk_logger.warning(f"{EXECUTOR_UNHEALTHY_MESSAGE}. Response: {response.status_code} - {response.text}")
+                return False
+            return True
+        except Exception as e:
+            bulk_logger.warning(f"{EXECUTOR_UNHEALTHY_MESSAGE}. Error: {str(e)}")
+            return False
+
+    def _check_startup_error_from_file(self, error_file) -> Exception:
+        error_dict = load_json(error_file)
```

Comment:
Do we need to handle the case if error_file is not found?

Reply:
I think the error file should be maintained by the executor proxy itself, both on creation and deletion.

Reply:
I'll add file deletion in this function.

If error_file is not found, I think it's a system error, so we don't need to handle this case.